MoreLinq.Test/IgnoreErrorsTest.cs

```diff
@@ -0,0 +1,92 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+    using System;
+
+    [TestFixture]
+    public class IgnoreErrorsTest
+    {
+        [Test]
+        public void IgnoreErrorsIsLazy()
+        {
+            new BreakingSequence<int>().IgnoreErrors(BreakingFunc.Of<Exception, bool>());
+        }
+
+        [Test]
+        public void IgnoreErrors()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => throw new TestException("True"),
+                                             () => 2,
+                                             () => throw new TestException("False"),
+                                             () => 3);
+
+            var result = source.IgnoreErrors((TestException e) => bool.Parse(e.Message));
+            var expectations = Enumerable.Range(1, 2);
+
+            Assert.That(result.Take(2), Is.EqualTo(expectations));
+            Assert.Throws<TestException>(() => result.ElementAt(2));
+        }
+
+        [Test]
+        public void IgnoreErrorsNoExceptionTypeMatch()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => 2,
+                                             () => throw new TestException(),
+                                             () => 3);
+
+            var result = source.IgnoreErrors((ArgumentException e) => true);
+            var expectations = Enumerable.Range(1, 2);
+
+            Assert.That(result.Take(2), Is.EqualTo(expectations));
+            Assert.Throws<TestException>(() => result.ElementAt(2));
+        }
+
+        [Test]
+        public void IgnoreErrorsWithBaseException()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => throw new TestException(),
+                                             () => 2,
+                                             () => throw new Exception(),
+                                             () => 3,
+                                             () => throw new NullReferenceException(),
+                                             () => 4,
+                                             () => throw new ArgumentException(),
+                                             () => 5);
+
+            var result = source.IgnoreErrors((Exception e) => true);
+            var expectations = Enumerable.Range(1, 5);
+
+            Assert.That(result, Is.EqualTo(expectations));
+        }
+
+        [Test]
+        public void IgnoreErrorsWithParserFunction()
+        {
+            var source = "O,l,2,3,4,S,6,7,B,9".Split(',').Select(int.Parse);
+            var result = source.IgnoreErrors((FormatException ex) => true);
+            var expectations = new[] { 2, 3, 4, 6, 7, 9 };
```

Comment:
The blank line in-between `expectations` and the assertion confuses things. They belong together:

```c#
var expectations = Enumerable.Range(1, 5);
Assert.That(result, Is.EqualTo(expectations));
```

If the idea is to delineate the [AAA](http://wiki.c2.com/?ArrangeActAssert) parts of the test then the blank line should at least come between the _act_ and _assert_ parts, as in:

```c#
var result = source.IgnoreErrors((Exception e) => true);

var expectations = Enumerable.Range(1, 5);
Assert.That(result, Is.EqualTo(expectations));
```

Do we even need `expectations`? I would even simplify to just:

```c#
result.AssertSequenceEqual(1, 2, 3, 4, 5);
```

and alternatively:

```c#
source.IgnoreErrors((Exception e) => true)
      .AssertSequenceEqual(1, 2, 3, 4, 5);;
```

Review other tests for the same.